### PR TITLE
Channel api

### PR DESCRIPTION
**Pull request by ayushdevelop**
# Related Issue

- https://github.com/code-mancers/g-bot/issues/14

# Functionalities added or changed

- created an api endpoint for fetching the channels of slack workspace

# Checklist

- [x] Tests
- [x] Uploaded screenshots

# Screenshots

Screenshot 2022-09-28 at 6 25 47 PM

### CODE REVIEW

File: server/src/__tests__/channels.test.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import request from "supertest";
+import expressApp from "../server";
+import { getChannels } from "../services/channels.service";
+import mockChannelData from "../services/__mock__/channel.mock";
```
Comment (SujayPrabhu96) [inline server/src/__tests__/channels.test.ts:4]: `__mocks__` folder can be placed inside `/server` rather than in `/services`

**Review by SujayPrabhu96 — APPROVED** [review]
One minor comment. Can be fixed in coming PRs. Otherwise LGTM

File: server/src/__tests__/channels.test.ts (hunk 1 above)
Comment (ayushdevelop) [reply server/src/__tests__/channels.test.ts:4]: Okay, will move it

File: server/src/__tests__/channels.test.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+jest.mock("../services/channels.service");
+
+const mockGetChannels = getChannels as jest.MockedFunction<typeof getChannels>;
+
+describe("GET", () => {
+  describe("/channels", () => {
```
Comment (SujayPrabhu96) [inline server/src/__tests__/channels.test.ts:11]:
```
describe("Channels", () => {
   describe("GET /channels", () => {
      /// tests
   })
})
```

**Pull request merged**